core/src/main/scala/kafka/coordinator/group/CoordinatorLoaderImpl.scala

```diff
@@ -148,6 +156,16 @@ class CoordinatorLoaderImpl[T](
                 }
               }
 
+              val currentHighWatermark = log.highWatermark
+              if (currentHighWatermark > previousHighWatermark) {
+                onHighWatermarkUpdated.accept(currentHighWatermark)
+                previousHighWatermark = currentHighWatermark
+              }
```

Comment:
I wonder if we should rather do this after updating the last written offset. At least, conceptually it makes more sense. What do you think?

Reply:
yeah, that makes sense to me since updating hwm also deletes snapshots